core/src/test/scala/unit/kafka/tools/DumpLogSegmentsTest.scala

```diff
@@ -243,6 +246,209 @@ class DumpLogSegmentsTest {
+    val expectedDeletePayload = String.format("RemotePartitionDeleteMetadata{topicPartition=%s:%s-0, " +
+      "state=DELETE_PARTITION_MARKED, eventTimestampMs=0, brokerId=0}", topicId, topicName)
+
+    val output = runDumpLogSegments(Array("--remote-log-metadata-decoder", "--files", logPaths(1)))
+    assertTrue(output.contains("Log starting offset: 1"))
+    assertTrue(output.contains(expectedDeletePayload))
+  }
+
+  @Test
+  def testDumpRemoteLogMetadataWithCorruption(): Unit = {
+    val metadataRecords = Array(new SimpleRecord(null, "corrupted".getBytes()))
+
+    val logConfig = LogTestUtils.createLogConfig(segmentBytes = 1024 * 1024)
+    log = LogTestUtils.createLog(logDir, logConfig, new BrokerTopicStats, time.scheduler, time)
+    log.appendAsLeader(MemoryRecords.withRecords(Compression.NONE, metadataRecords:_*), leaderEpoch = 0)
+    log.flush(false)
+
+    val output = runDumpLogSegments(Array("--remote-log-metadata-decoder", "--files", logFilePath))
+    assertTrue(output.contains("Log starting offset: 0"))
+    assertTrue(output.contains("Could not deserialize metadata record"))
+  }
+
+  @Test
+  def testDumpRemoteLogMetadataIoException(): Unit = {
+    val topicId = Uuid.randomUuid
+    val topicName = "foo"
+
+    val metadata = Seq(new RemotePartitionDeleteMetadata(new TopicIdPartition(topicId, new TopicPartition(topicName, 0)),
+      RemotePartitionDeleteState.DELETE_PARTITION_MARKED, time.milliseconds, 0))
+
+    val metadataRecords: Array[SimpleRecord] = metadata.map(message => {
+      new SimpleRecord(null, new RemoteLogMetadataSerde().serialize(message))
+    }).toArray
+    
+    val logConfig = LogTestUtils.createLogConfig(segmentBytes = 1024 * 1024)
+    log = LogTestUtils.createLog(logDir, logConfig, new BrokerTopicStats, time.scheduler, time)
+    log.appendAsLeader(MemoryRecords.withRecords(Compression.NONE, metadataRecords:_*), leaderEpoch = 0)
+    log.flush(false)
+    
+    Files.setPosixFilePermissions(Paths.get(logFilePath), PosixFilePermissions.fromString("-w-------"))
+    
+    assertThrows(classOf[AccessDeniedException],
+      () => runDumpLogSegments(Array("--remote-log-metadata-decoder", "--files", logFilePath)))
+  }
+
+  @Test
+  def testDumpRemoteLogMetadataNoFilesFlag(): Unit = {
+    val topicId = Uuid.randomUuid
+    val topicName = "foo"
+
+    val metadata = Seq(new RemotePartitionDeleteMetadata(new TopicIdPartition(topicId, new TopicPartition(topicName, 0)),
+      RemotePartitionDeleteState.DELETE_PARTITION_MARKED, time.milliseconds, 0))
+
+    val records: Array[SimpleRecord] = metadata.map(message => {
+      new SimpleRecord(null, new RemoteLogMetadataSerde().serialize(message))
+    }).toArray
+
+    val logConfig = LogTestUtils.createLogConfig(segmentBytes = 1024 * 1024)
+    log = LogTestUtils.createLog(logDir, logConfig, new BrokerTopicStats, time.scheduler, time)
+    log.appendAsLeader(MemoryRecords.withRecords(Compression.NONE, records:_*), leaderEpoch = 0)
+    log.flush(false)
```

Comment:
Is this needed?

Reply:
Why not? Without flushing the dump tool risks to not find appended records on disk right?

Reply:
No, I mean the whole preparation part. This test is testing `NoFilesFlag`, which means we don't need any log files. Is that right?

Reply:
Ah sorry, you are right :)